iOS/ViewControllers/UpcomingMoviesViewController.swift

```diff
@@ -0,0 +1,98 @@
+import UIKit
+import Shared
+
+protocol ListUpcomingMoviesDelegate: class {
+
+    func didList(movies: [Movie])
+    func didLoad(genres: [Genre])
+    func didDisplayError(message: String)
+    func didGetAvatar(movie: Movie, image: UIImage)
+
+}
+
+class UpcomingMoviesViewController: UIViewController, ListUpcomingMoviesDelegate {
+
+    private var cellIdentifier = String(describing: MovieTableViewCell.self)
+    private var tableViewDataSource: GenericTableViewDataSource<Movie, MovieTableViewCell>?
+    private var tableViewDelegate: GenericTableViewDelegate?
+    private var listUpcomingMoviesInteractor: ListUpcomingMoviesInteractor?
+    private var getMovieBackdropInteractor: GetMovieBackdropInteractor?
+    private var genres: [Genre] = []
+
+    @IBOutlet weak var tableView: UITableView!
+
+    override func viewDidLoad() {
+        configureGetMovieBackdropInteractor()
+        configureTableView()
+        configureUpcomingMoviesInteractor()
+        super.viewDidLoad()
+    }
+
+    private func configureGetMovieBackdropInteractor() {
+        getMovieBackdropInteractor = GetMovieBackdropInteractorFactory.make(
+            presenter: GetMovieBackdropPresenteriOS(delegate: self))
+    }
+
+    private func configureTableView() {
+        let cellNib = UINib(nibName: cellIdentifier, bundle: nil)
+        tableView.register(cellNib, forCellReuseIdentifier: cellIdentifier)
+        configureTableViewDataSource()
+        configureTableViewDelegate()
+    }
+
+    private func configureTableViewDataSource() {
+        tableViewDataSource = GenericTableViewDataSource() { (movie, cell) in
+            cell.configure(movie: movie)
+            let movieGenres = self.genres.filter({ movie.genresIds.contains($0.id) })
+            cell.configure(genres: movieGenres)
+            guard let dataSource = self.tableViewDataSource, let image = dataSource.images[movie.id] else {
+                self.getMovieBackdropInteractor?.get(movie: movie)
```

Comment:
When the image of movie was not in the cached dictionary the image will be requested. After request they will be rendered at `didGetAvatar` https://github.com/ronanrodrigo/Cinefilia/pull/1/files/7aa51d5500ff7b6b709e1659ce7e0e34cc71c6da#diff-ef2b607d4e0f9b3d22967ef664d5d876R87

If it was in dictionary, just pas it to `cell.configure(image: image)`